pulpcore/cli/rpm/content.py

```diff
@@ -52,74 +65,215 @@ def _sha256_artifact_callback(
 
 
 @pulp_group()
-@click.option(
-    "-t",
-    "--type",
-    "content_type",
-    type=click.Choice(["package"], case_sensitive=False),
+@type_option(
+    choices={
+        "package": PulpRpmPackageContext,
+        "advisory": PulpRpmAdvisoryContext,
+        "distribution_tree": PulpRpmDistributionTreeContext,
+        "modulemd_defaults": PulpRpmModulemdDefaultsContext,
+        "modulemd": PulpRpmModulemdContext,
+        "package_category": PulpRpmPackageCategoryContext,
+        "package_environment": PulpRpmPackageEnvironmentContext,
+        "package_group": PulpRpmPackageGroupContext,
+        "package_langpack": PulpRpmPackageLangpacksContext,
+        "repo_metadata_file": PulpRpmRepoMetadataFileContext,
+    },
     default="package",
+    case_sensitive=False,
 )
 @pass_pulp_context
 @click.pass_context
-def content(ctx: click.Context, pulp_ctx: PulpContext, content_type: str) -> None:
-    if content_type == "package":
-        ctx.obj = PulpRpmPackageContext(pulp_ctx)
-    else:
-        raise NotImplementedError()
+def content(ctx: click.Context, pulp_ctx: PulpContext) -> None:
```

Comment:
At this point you don't even need to use the pass decorators anymore.

Reply:
Concur. First cut at this PR was in mid-experiment state, pushed specifically so I could get more eyes on it.